core/src/sigverify_stage.rs

```diff
@@ -248,20 +248,27 @@ impl SigVerifyStage {
             num_packets,
         );
 
-        let mut dedup_time = Measure::start("sigverify_dedup_time");
-        let dedup_fail = deduper.dedup_packets(&mut batches) as usize;
-        dedup_time.stop();
-        let num_unique = num_packets.saturating_sub(dedup_fail);
+        use rand::{thread_rng, Rng};
+        const MAX_BATCHES: usize = 2000;
+        if batches.len() > MAX_BATCHES {
+            let percent_to_keep = (batches.len() - MAX_BATCHES) / batches.len();
+            batches.retain(|_batch| thread_rng().gen_ratio(percent_to_keep as u32, 100));
```

Comment:
It sounds to me like `percent_to_keep as u32` would always be 0? Needs a `100 * `